src/test/kotlin/com/example/delivery/controller/UserControllerTest.kt

```diff
@@ -0,0 +1,72 @@
+package com.example.delivery.controller
+
+import com.example.delivery.UserFixture.createUserDTO
+import com.example.delivery.UserFixture.domainUser
+import com.example.delivery.service.UserService
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import org.mockito.kotlin.doNothing
+import org.mockito.kotlin.verify
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.content
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status
+
+@WebMvcTest(UserController::class)
+internal class UserControllerTest {
+    @MockBean
+    private lateinit var userService: UserService
+
+    @Autowired
+    private lateinit var mockMvc: MockMvc
+
+    private val objectMapper = jacksonObjectMapper()
+
+    @Test
+    fun `should add user and return created`() {
+        //GIVEN
+        Mockito.`when`(userService.add(createUserDTO)).thenReturn(domainUser)
+
+        //THEN
+        mockMvc.perform(
+            post("/users")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(createUserDTO))
+        )
+            .andExpect(jsonPath("$.fullName").value("FULL NAME"))
+            .andExpect(jsonPath("$.phone").value("+31243123"))
+    }
+
+    @Test
+    fun `should return user when user exists`() {
+        //GIVEN
+        Mockito.`when`(userService.findById("1")).thenReturn(domainUser)
+
+        //THEN
+        mockMvc.perform(get("/users/{id}", "1"))
+            .andExpect(status().isOk)
+            .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(jsonPath("$.fullName").value("FULL NAME"))
+            .andExpect(jsonPath("$.phone").value("+31243123"))
+    }
+
+    @Test
+    fun `should delete existing user and return no content`() {
+        //GIVEN
+        doNothing().`when`(userService).deleteById("1")
+
+        //THEN
+        mockMvc.perform(delete("/users/{id}", "1"))
+            .andExpect(status().isNoContent)
+
+        verify(userService).deleteById("1")
```

Comment:
```suggestion
        // WHEN // THEN
        mockMvc.perform(delete("/users/{id}", "1"))
            .andExpect(status().isNoContent)

        // AND THEN
        verify(userService).deleteById("1")
```

- `given` is setup block
- `when` is actual action to test
- `then` is assertion block

Sometimes `when` and `then` block shouldn't be splitted(as in your case here) so we can use `// WHEN // THEN` or `// WHEN-THEN` comment.